packages/eslint-plugin/lib/rules/unified-signatures.js

```diff
@@ -0,0 +1,574 @@
+                    if (statement.type === Syntax.TSDeclareFunction) {
+                        const { body, id } = statement;
+                        return body === undefined && id !== undefined
+                            ? { signature: statement, key: id.name }
+                            : undefined;
+                    } else {
+                        return undefined;
+                    }
+                })
+            );
+        }
+
+        /**
+         * @param {Array} members
+         * @param {Array} [typeParameters]
+         * @returns {void}
+         */
+        function checkMembers(members, typeParameters) {
+            addFailures(
+                checkOverloads(members, typeParameters, member => {
+                    switch (member.type) {
+                        case Syntax.TSCallSignatureDeclaration:
+                        case Syntax.TSConstructSignatureDeclaration:
+                        case Syntax.TSMethodSignature:
+                            break;
+                        case Syntax.TSAbstractMethodDefinition:
+                        case Syntax.MethodDefinition:
+                            if (member.value.body !== null) {
+                                return undefined;
+                            }
+                            break;
+                        default:
+                            return undefined;
+
+                    }
+
+                    const key = getOverloadKey(member);
+                    return key === undefined ? undefined : { signature: member, key };
+                })
+            );
+        }
+
+        /**
+         * @param {Failure[]} failures
+         */
+        function addFailures(failures) {
+            for (const failure of failures) {
+                const { unify, only2 } = failure;
+                switch (unify.kind) {
+                    case "single-parameter-difference": {
+                        const { p0, p1 } = unify;
+                        const lineOfOtherOverload = only2 ? undefined : p0.loc.start.line
+                        context.report({
+                            loc: p1.loc,
+                            message: FAILURE_STRING_SINGLE_PARAMETER_DIFFERENCE(
+                                lineOfOtherOverload,
+                                sourceCode.getText(p0.typeAnnotation.typeAnnotation),
+                                sourceCode.getText(p1.typeAnnotation.typeAnnotation)
+                            ),
+                        });
+                        break;
+                    }
+                    case "extra-parameter": {
+                        const { extraParameter, otherSignature } = unify;
+                        const lineOfOtherOverload = only2 ? undefined : otherSignature.loc.start.line;
+
+                        context.report({
+                            loc: extraParameter.loc,
+                            message: extraParameter.type === Syntax.RestElement
```

Comment:
can you convert `message` to `messageId` with `data`
you can find example here: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/lib/rules/no-type-alias.js#L32

https://eslint.org/docs/developer-guide/working-with-rules#messageids

Reply:
Converted.